chainer/iterators/multiprocess_iterator.py

```diff
@@ -62,20 +87,20 @@ class MultiprocessIterator(iterator.Iterator):
 
     _interruption_testing = False  # for testing
     _finalized = False
+    _prefetch_loop = None
     _comm = None
-    _thread = None
 
     def __init__(self, dataset, batch_size, repeat=True, shuffle=None,
                  n_processes=None, n_prefetch=1, shared_mem=None,
-                 order_sampler=None):
+                 order_sampler=None, dataset_timeout=30.0):
```

Comment:
I think this change is no compatible. How about you?

Reply:
In which case do you think it's no-compat?

Reply:
The current implementation does not cause a timeout.
Is a 30 second timeout fine?

Reply:
It depends on how user dataset is implemented.
For example, it could read a file with stale nfs handle.

I think 30 seconds is reasonable.
Usually dataset realization does not take a second. 30 seconds is a reasonable time for human to run watch-for-error-and-fix loop.

Reply:
Some datasets like `ZippedImageDataset` requires initialization at first access; opening the whole ZIP file may take seconds. How about this idea?
* By default, show warning in the parent process (instead of raising an exception) when stall is detected.
* Introduce a bool flag to raise an exception instead of warning.

Reply:
I think that sounds good.

Reply:
I fixed as such.
I didn't introduce the flag, because users can turn the warning into an error by using `warnings.filterwarnings`.

Reply:
OK, Thanks